chaos_genius/connectors/aws_athena.py

```diff
@@ -0,0 +1,140 @@
+        aws_secret_access_key = db_info.get("aws_secret_access_key")
+        region_name = db_info.get("region_name")
+        schema_name = db_info.get("schema_name", "default")
+        s3_staging_dir = db_info.get("s3_staging_dir")
+        if not (
+            aws_access_key_id
+            and aws_secret_access_key
+            and region_name
+            and s3_staging_dir
+        ):
+            raise NotImplementedError(
+                "Database Credential not found for AWS Athena."
+            )
+
+        conn_string = "awsathena+rest://{aws_access_key_id}:{aws_secret_access_key}@athena.{region_name}.amazonaws.com/{schema_name}?s3_staging_dir={s3_staging_dir}".format(
+            aws_access_key_id=quote_plus(aws_access_key_id),
+            aws_secret_access_key=quote_plus(aws_secret_access_key),
+            region_name=region_name,
+            schema_name=schema_name,
+            s3_staging_dir=quote_plus(s3_staging_dir),
+        )
+        self.sqlalchemy_db_uri = conn_string
+        return self.sqlalchemy_db_uri
+
+    def get_db_engine(self):
+        db_uri = self.get_db_uri()
+        self.engine = create_engine(db_uri, echo=self.debug)
+        return self.engine
+
+    def get_pyathena_engine(self):
+        db_info = self.ds_info
+        aws_access_key_id = db_info.get("aws_access_key_id")
+        aws_secret_access_key = db_info.get("aws_secret_access_key")
+        region_name = db_info.get("region_name")
+        s3_staging_dir = db_info.get("s3_staging_dir")
+        if not (
+            aws_access_key_id
+            and aws_secret_access_key
+            and region_name
+            and s3_staging_dir
+        ):
+            raise NotImplementedError(
+                "Database Credential not found for AWS Athena."
+            )
+        cursor = connect(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            s3_staging_dir=s3_staging_dir,
+            region_name=region_name,
+            cursor_class=PandasCursor,
+        ).cursor()
+        return cursor
+
+    def test_connection(self):
+        if not hasattr(self, "engine") or not self.engine:
+            self.engine = self.get_db_engine()
+        query_text = text(self.test_db_query)
+        status, message = None, ""
+        try:
+            with self.engine.connect() as connection:
+                cursor = connection.execute(query_text)
+                results = cursor.all()
+                if results[0][0] == 1:
+                    status = True
+                else:
+                    status = False
+        except Exception as err_msg:
+            status = False
+            message = str(err_msg)
+        return status, message
+
+    def run_query(self, query, as_df=True):
+        cursor = self.get_pyathena_engine()
+        if as_df == True:
+            # TODO: Test the performance
+            return cursor.execute(query).as_pandas()
```

Comment:
Can you check the possibilities of chunking here?

Reply:
done